include/deal.II/base/polynomials_rt_bubbles.h

```diff
@@ -0,0 +1,186 @@
+ * space dimension. This allows to associate shape functions with the Gauss-Lobatto quadrature
+ * points as shown in the figures below.
+ *
+ * <table> <tr> <td align="center">
+ * @image html http://www.dealii.org/images/shape-functions/RTBubbles/rtbubbles.png
+ * </td>
+ *
+ * <td align="center"> </td> </tr> <tr> <td align="center"> Left - $2D,\,k=3$,
+ * right - $3D,\,k=2$.</td>
+ *
+ * <td align="center"></tr> </table>
+ *
+ * @ingroup Polynomials
+ * @author Eldar Khattatov
+ * @date 2018
+ */
+
+template <int dim>
+class PolynomialsRT_Bubbles
+{
+public:
+  /**
+   * Constructor. Creates all basis functions for RT_bubbles polynomials of given
+   * degree.
+   */
+  PolynomialsRT_Bubbles (const unsigned int k);
+
+  /**
+   * Computes the value and the first and second derivatives of each RT_bubbles
+   * polynomial at @p unit_point.
+   *
+   * The size of the vectors must either be zero or equal <tt>n()</tt>.  In
+   * the first case, the function will not compute these values.
+   *
+   * If you need values or derivatives of all tensor product polynomials then
+   * use this function, rather than using any of the <tt>compute_value</tt>,
+   * <tt>compute_grad</tt> or <tt>compute_grad_grad</tt> functions, see below,
+   * in a loop over all tensor product polynomials.
+   */
+  void compute (const Point<dim>            &unit_point,
+                std::vector<Tensor<1,dim> > &values,
+                std::vector<Tensor<2,dim> > &grads,
+                std::vector<Tensor<3,dim> > &grad_grads,
+                std::vector<Tensor<4,dim> > &third_derivatives,
+                std::vector<Tensor<5,dim> > &fourth_derivatives) const;
+
+  /**
+   * Returns the number of enhanced Raviart-Thomas polynomials.
+   */
+  unsigned int n () const;
+
+  /**
+   * Returns the degree of the RT_bubble space, which is one less than the highest
+   * polynomial degree.
+   */
+  unsigned int degree () const;
+
+  /**
+   * Return the name of the space, which is <tt>RT_Bubbles</tt>.
+   */
+  std::string name () const;
+
+  /**
+   * Return the number of polynomials in the space <tt>RT_Bubbles(degree)</tt>
+   * without requiring to build an object of PolynomialsRT-Bubbles. This is required
+   * by the FiniteElement classes.
+   */
+  static unsigned int compute_n_pols(const unsigned int degree);
+
+private:
+  /**
+   * The degree of this object as given to the constructor.
+   */
+  const unsigned int my_degree;
+
+  /**
+  * An object representing the Raviart-Thomas part of the space
+  */
+  const PolynomialsRaviartThomas<dim> raviart_thomas_space;
+
+  /**
+   * Storage for monomials, we need all polynomials from degree zero
+   * to <i>k+1</i>.
+   */
+  std::vector<Polynomials::Polynomial<double>> monomials;
+
+  /**
+   * Number of RT_Bubbles polynomials.
+   */
+  unsigned int n_pols;
+};
+
+
+template <int dim>
+inline unsigned int
+PolynomialsRT_Bubbles<dim>::n() const
+{
+  return n_pols;
+}
+
+
```

Comment:
We use three blank lines between functions. Can you change here and below when necessary.